src/map/script-fun.cpp

```diff
@@ -788,7 +788,9 @@ void builtin_requestitem(ScriptState *st)
     dumb_ptr<map_session_data> sd = nullptr;
     script_data& scrd = AARG(0);
     assert (scrd.is<ScriptDataVariable>());
-    int amount = HARG(1) ? conv_num(st, &AARG(1)) : 0;
+    int amount = HARG(1) ? conv_num(st, &AARG(1)) : 1;
+    if (amount < 1 || amount > 16)
```

Comment:
here I put 16 as max because requiring the player to drag&drop +16 items is just insane (and 16 is already generous)